### PR TITLE
feat: add a default fuzz test seed when none is provided

### PR DESCRIPTION
**Pull request by schaable**
Closes https://github.com/NomicFoundation/hardhat/issues/7863

### CODE REVIEW

File: v-next/hardhat/src/internal/builtin-plugins/solidity-test/config.ts [hunk 1]
```diff
@@ -169,3 +175,12 @@ export async function resolveSolidityTestUserConfig(
     },
   };
 }
+
+export function resolveFuzzConfig(
```
Comment (ChristopherDedominici) [inline v-next/hardhat/src/internal/builtin-plugins/solidity-test/config.ts:179]: Not a blocker, but I’m wondering about the following:

the type `SolidityTestConfig["fuzz"]` defines `seed` as `string | undefined`. Since `seed` is always defined here, I’m wondering whether enforcing it as defined would improve the developer experience.

I don’t have full knowledge of the gas feature, so this may not be very impactful. Feel free to disregard this comment in that case

**Review by ChristopherDedominici — APPROVED** [review]
I left a comment, but other than that LGTM

File: v-next/hardhat/src/internal/builtin-plugins/solidity-test/config.ts (hunk 1 above)
Comment (schaable) [reply v-next/hardhat/src/internal/builtin-plugins/solidity-test/config.ts:179]: I agree, good catch!

Comment (schaable) [reply v-next/hardhat/src/internal/builtin-plugins/solidity-test/config.ts:179]: Done [here](https://github.com/NomicFoundation/hardhat/pull/7886/commits/b191531ee42a810b41ccd3b6c50bb27d2d513976). @kanej this is the change we discussed on Slack. It ended up being a bit more complex than expected, since the fuzz config is now always resolved to an object containing a seed string. I’m going to merge this into the feature branch. Let me know if you want me to tweak the commit in any way.
